src/pydap/responses/dods.py

```diff
@@ -203,18 +204,26 @@ def _basetype(var):
         if DAP2_dtype.char == 'S':
             for block in data:
                 for word in block.flat:
-                    length = len(word)
-                    yield tostring_with_byteorder(
-                                np.array(length),
-                                np.dtype(DAP2_ARRAY_LENGTH_NUMPY_TYPE))
                     # byteorder is not important for strings:
-                    if hasattr(word, 'encode'):
-                        yield word.encode('ascii')
-                    elif hasattr(word, 'tostring'):
-                        yield word.tostring()
-                    else:
-                        raise TypeError("Could not convert word '{0}' to bytes"
-                                        .format(word))
+                    try:
+                        enc = quote_val(word)
+                        length = len(enc)
+                        yield tostring_with_byteorder(
+                                    np.array(length),
+                                    np.dtype(DAP2_ARRAY_LENGTH_NUMPY_TYPE))
+                        yield enc
+                    except:
```

Comment:
Again here, what is your try clause capturing? It is too wide and "yield tostring_wit..." is repeated. This must be avoided.

Reply:
Echo @laliberte, need to be more explicit here.